### PR TITLE
add quant precision

### PR DESCRIPTION
**Pull request by Sarahb-da**
Adds quantity precision check to splitting assets

### CODE REVIEW

File: ui/src/components/common/Holdings.tsx [hunk 1]
```diff
@@ -207,36 +213,39 @@ const SplitForm: React.FC<SplitFormProps> = ({ deposit }) => {
...
 
+    const validateSplitNumber = (event: React.SyntheticEvent, result: any) => {
+        const number = Number(result.value)
+
+        if (number >= Number(asset.quantity)) {
+            return setSplitNumberError(`The splitting quantity must be less than ${asset.quantity}`)
+        }
+
+        if (number < 0) {
+            return setSplitNumberError(`The splitting quantity must be greater than 0.`)
```
Comment (dimitri-da) [inline ui/src/components/common/Holdings.tsx:230]: here the error message does not agree with the check. I think it should be `<=0`

File: ui/src/components/common/Holdings.tsx [hunk 2]
```diff
@@ -207,36 +213,39 @@ const SplitForm: React.FC<SplitFormProps> = ({ deposit }) => {
...
+        }
+
+        if (countDecimals(number) > tokenQuantityPercision) {
+            return setSplitNumberError(`The decimal percision of the splitting quantity must be equal to or less than ${tokenQuantityPercision}.`)
```
Comment (dimitri-da) [inline ui/src/components/common/Holdings.tsx:234]: typo `precision`

File: ui/src/components/common/Holdings.tsx [hunk 3]
```diff
@@ -199,6 +200,11 @@ type SplitFormProps = {
 const SplitForm: React.FC<SplitFormProps> = ({ deposit }) => {
     const { asset } = deposit.contractData;
     const ledger = useLedger();
+    const [ splitNumberError, setSplitNumberError ] = useState<string>()
+
+    const tokenQuantityPercision = Number(useStreamQuery(Token).contracts
```
Comment (dimitri-da) [inline ui/src/components/common/Holdings.tsx:205]: if the websocket fails here do we crash and burn? should we have a default precision or show some error?

File: ui/src/components/common/utils.ts [hunk 4]
```diff
@@ -45,3 +45,9 @@ export const depositSummary = (deposits: DepositInfo[]): string => {
         .map(([label, total]) => `${label}: ${total}`)
         .join(", ");
 }
+
+export function countDecimals(value: number) {
+    if (Math.floor(value) !== value)
+        return value.toString().split(".")[1].length || 0;
```
Comment (dimitri-da) [inline ui/src/components/common/utils.ts:51]: if my split quantity is `1.2000000000000000000000000000000000` will this count for 1 decimal or 34? Not really a big deal as I can always remove the trailing zeros but just wondering

**Review by dimitri-da — COMMENTED** [review]
found a typo and a couple of minor comments

File: ui/src/components/common/utils.ts (hunk 4 above)
Comment (Sarahb-da) [reply ui/src/components/common/utils.ts:51]: the `toString()` method strips trailing zeros, so the count would be 1

File: ui/src/components/common/Holdings.tsx (hunk 3 above)
Comment (Sarahb-da) [reply ui/src/components/common/Holdings.tsx:205]: yep will add fo rthis

**Review by dimitri-da — APPROVED** [review]

**Pull request merged**